### ISSUE TITLE
[❔ other question] How to parallelly project image 

### ISSUE THREAD

**Issue by atuyoti** [mitsuba-renderer/mitsuba2#425]
I'm trying to 3D reconstruction.
To do that, I need the parallel projection.
However, emitter type="projector" is the perspective projection.
Are there any suitable methods for the parallel projection?

**Comment by merlinND** [comment]
Hello @atuyoti,

If I understand correctly, you are looking for an emitter that can be textured and emits radiance in parallel rays.
This could be achieved with a `directionalarea` emitter attached to a simple rectangle shape.

https://github.com/mitsuba-renderer/mitsuba2/blob/7c9c5391d83d8499e52ce90b6763e00abd9a0633/src/emitters/directionalarea.cpp

It is not merged yet to `master` but should be usable (you might have to remove methods that are not needed on `master`).

**Comment by atuyoti** [comment]
Thank you @merlinND 
Your understanding is correct.

I tried to using directionalarea emitter.

```
  <shape type="rectangle" id="rect">
    <transform name="to_world">
      <rotate y="1" angle="90"/>
      <translate x="0" y="0" z="5"/>
    </transform>
  </shape>

  <emitter type="directionalarea">
    <ref id="rect"/>
    <texture name="reflectance" type="bitmap">
      <string name="filename" value="./projectionimage3/proimage_000.png"/>
    </texture>
  </emitter>
```

Is it correct?
In addition, when I run python code and use directionalarea emitter, An error occurred.
```
  File "/Users/itouyuuta/Documents/GitHub/mitsuba2/docs/examples/fog_render/render_DirectionalArea.py", line 28, in <module>
    scene = load_file(filename)
RuntimeError: ​[xml.cpp:1047] Error while loading "test_DirectionalArea.xml" (near line 34, col 4): could not instantiate emitter plugin of type "directionalarea": 
[xml.cpp:1047]   [PluginManager] Plugin "directionalarea" not found!
```
The reason I think is branch, so I changed branch from master to the branch where are directionalarea.cpp.
As a result, build was not successful.
error message
```
../src/emitters/directionalarea.cpp:56:37: error: use of undeclared identifier 'ek'
        SurfaceInteraction3f si(ps, ek::zero<Wavelength>());

../src/emitters/directionalarea.cpp:89:40: error: only virtual member functions can be marked 'override'
                    Mask active) const override {
```

**Comment by merlinND** [comment]
Right, that plugin is not yet available on `master` and the corresponding branch is not ready yet for merging (not tested on different platforms and modes, etc).

If you wish to use the `broken-ptracer` branch, make sure that you update all submodules recursively and do a clean build.
Another option would be to copy the `directionalarea` plugin to your `master` branch, and make the small required fixes for it to compile. Any extra methods that are not part of the `Emitter` interface on `master` can be removed.

Unfortunately, until I am able to complete work on that branch and merge it properly, I will not be able to support these specific use-cases. I hope the existing code should be a helpful starting point though.
